### PR TITLE
Use NonEmpty for SenderSignedData

### PR DESCRIPTION
**Pull request by mystenmark**
Removes the need for several asserts, since all failures will now be deserialization errors

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -382,7 +382,7 @@ msim-macros = { git = "https://github.com/MystenLabs/mysten-sim.git", rev = "6f8
 multiaddr = "0.17.0"
 nexlint = { git = "https://github.com/nextest-rs/nexlint.git", rev = "94da5c787636dad779c340affa65219134d127f5" }
 nexlint-lints = { git = "https://github.com/nextest-rs/nexlint.git", rev = "94da5c787636dad779c340affa65219134d127f5" }
-nonempty = "0.9.0"
+nonempty = { verison = "0.9.0", features = ["serialize"]}
```
Comment (bmwill) [inline Cargo.toml:385]: Is this change still needed?

**Review by bmwill — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (mystenmark) [reply Cargo.toml:385]: no!

**Comment by lxfind** [comment]
Very cool!
